### DISCUSSION TITLE
When a `ConnectionPool` eliminate broken connection?

### DISCUSSION THREAD

**Discussion by JackyWoo** [ClickHouse/ClickHouse#42948] [Q&A]
I search the code but did not find it.

**Comment by mo-avatar** [comment]
To my knowledge ConnectionPool doesn't literally eliminate broken connection, it only set the in_use flag  of PooledObject in PoolBase.h to false  , when entry dtor executed(more specifically the PoolEntryHelper dtor executed).

**Reply by JackyWoo** [reply]
Thanks. `ConnectionPool` never shrink?

**Reply by mo-avatar** [reply]
To my knowledge, yes, it never shrink.
